fasthttpproxy/proxy_env.go

```diff
@@ -0,0 +1,108 @@
+package fasthttpproxy
+
+import (
+	"bufio"
+	"encoding/base64"
+	"fmt"
+	"net"
+	"net/url"
+	"sync"
+	"time"
+
+	"golang.org/x/net/http/httpproxy"
+
+	"github.com/valyala/fasthttp"
+)
+
+// FasthttpProxyHTTPDialer returns a fasthttp.DialFunc that dials using
+// the the env(HTTP_PROXY, HTTPS_PROXY and NO_PROXY) configured HTTP proxy.
+//
+// Example usage:
+//	c := &fasthttp.Client{
+//		Dial: FasthttpProxyHTTPDialer(),
+//	}
+func FasthttpProxyHTTPDialer() fasthttp.DialFunc {
+	return FasthttpProxyHTTPDialerTimeout(0)
+}
+
+// FasthttpProxyHTTPDialer returns a fasthttp.DialFunc that dials using
+// the env(HTTP_PROXY, HTTPS_PROXY and NO_PROXY) configured HTTP proxy using the given timeout.
+//
+// Example usage:
+//	c := &fasthttp.Client{
+//		Dial: FasthttpProxyHTTPDialerTimeout(time.Second * 2),
+//	}
+func FasthttpProxyHTTPDialerTimeout(timeout time.Duration) fasthttp.DialFunc {
+	proxier := httpproxy.FromEnvironment().ProxyFunc()
+
+	// map on proxy URL and its encoded auth barrier
+	authBarriers := map[*url.URL]string{}
+	authBarriersLock := sync.RWMutex{}
+
+	return func(addr string) (net.Conn, error) {
+
+		proxyURL, err := proxier(&url.URL{Host: addr})
```

Comment:
I'm wondering if `Scheme` should be set to `https` here if the port is `443`?

Reply:
Did you test this code? Not setting `Scheme` makes this always return `nil`.
See: https://github.com/golang/net/blob/08b38378de702b893ee869b94b32f833e2933bd2/http/httpproxy/proxy.go#L129-L138

Reply:
@erikdubbelboer Hi! thanks for the review. You are pointing to the outdated version of the x/net/httpproxy. 
here is which one is currently used: https://github.com/golang/net/blob/627f9648deb96c27737b83199d44bb5c1010cbcf/http/httpproxy/proxy.go#L130-L138